h/accounts/views.py

```diff
@@ -132,7 +132,24 @@ def _logout(self):
 class AjaxAuthController(AuthController):
     def login(self):
         try:
-            appstruct = self.form.validate(self.request.json_body.items())
+            json_body = self.request.json_body
+        except ValueError as err:
+            self.request.status_code = 400
+            error = 'Could not parse request body as JSON: ' + err.message
+            return ajax_payload(
+                self.request, {'status': 'failure', 'errors': error})
```

Comment:
The `errors` key elsewhere is a dictionary mapping field names to error strings. I think this should probably be `reason`.